hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -411,9 +413,23 @@ class Setting(object):
     deprecated_since = attr.ib()
 
 
+def _max_examples_validator(x):
+    check_type(integer_types, x)
```

Comment:
Depending on how pedantic we want to be with backwards compatibility (and I'm inclined to think the answer is "fairly"), this might want to be a `try_convert`, possibly with a deprecation warning for non-integers.